sky/provision/gcp/instance_utils.py

```diff
@@ -75,6 +75,15 @@ def filter(
     ) -> List[str]:
         raise NotImplementedError
 
+    @classmethod
+    def get_vpc_name(
+        cls,
+        project_id: str,
+        zone: str,
+        instance: str,
+    ) -> Optional[str]:
+        raise NotImplementedError
+
```

Comment:
Again, do we need to add this new API, or should we just let the `create_or_update_firewall_rule()` to handle this?

Reply:
Because two types of instances (`ComputeInstance` and `TPUVMInstance`) have different functions to retrieve the vpc name, so we need an override here

Reply:
I mean why don't we do the `get_vpc_name` in the `create_or_update_firewall_rule()`? Can't we avoid exposing this function

Reply:
Because we cannot add a `create_or_update_firewall_rule` function in `TPUVMInstance` (it requires compute resources, i.e. the `load_resources` function in `ComputeInstance`), but we still need to add a firewall rule for a TPU VM cluster

Reply:
Ahh, I see. That makes sense. Thanks!